### PR TITLE
ISPN-5085 Switch Resteasy backend to Netty

### PR DESCRIPTION
**Pull request by galderz**
* Undertow web no longer used for serving REST requests, instead
use Netty based Resteasy backend.
* Removed all references to undertow, including dependencies and
subsystem configuration.
* Adjusted testsuite REST server tests so that they don't rely
on content length reading and instead they use streaming capabilities
to read as much as needed.
* Upgraded to Resteasy 3.0.11 version to get around an issue with
empty responses.

### CODE REVIEW

**Comment by galderz** [comment]
It should be pretty much complete except for security integration (whose tests are skipped currently), hence the preview label.

File: server/integration/endpoint/src/main/java/org/infinispan/server/endpoint/subsystem/RestSubsystemAdd.java [hunk 1]
```diff
@@ -58,13 +58,12 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
 
       String virtualServer = config.hasDefined(ModelKeys.VIRTUAL_SERVER) ? config.get(ModelKeys.VIRTUAL_SERVER).asString() : DEFAULT_VIRTUAL_SERVER;
```
Comment (tristantarrant) [inline server/integration/endpoint/src/main/java/org/infinispan/server/endpoint/subsystem/RestSubsystemAdd.java]: You need to setHostname() on the NettyJaxrsServer based on the virtual server name.

**Comment by galderz** [comment]
This is ready now

**Comment by tristantarrant** [comment]
I guess we lose the welcome page with this ? Any way we can serve it ?

**Comment by galderz** [comment]
@tristantarrant I'll check...

**Comment by galderz** [comment]
Talked to @tristantarrant and the welcome page is not crucial at this stage.

**Comment by tristantarrant** [comment]
Pushed, thanks @galderz

**Comment by gustavonalle** [comment]
Guys, CI is your friend, there were failing tests in this PR and still it was integrated :smile:

**Comment by tristantarrant** [comment]
I have a fix for this in https://github.com/infinispan/infinispan/pull/3386

**Pull request closed without merging**